libcxx/include/__tree

```diff
@@ -1276,6 +1267,32 @@ private:
   }
   _LIBCPP_HIDE_FROM_ABI void __move_assign_alloc(__tree&, false_type) _NOEXCEPT {}
 
+  template <class _From, __enable_if_t<__is_pair_v<__remove_cvref_t<_From> >, int> = 0>
+  _LIBCPP_HIDE_FROM_ABI static void __assign_value(__get_node_value_type_t<value_type>& __lhs, _From&& __rhs) {
+    using __punned_type = pair<typename _NodeTypes::key_type, typename _NodeTypes::mapped_type>;
+
+    reinterpret_cast<__punned_type&>(__lhs) = reinterpret_cast<__copy_cvref_t<_From, __punned_type>&&>(__rhs);
```

Comment:
Hmm, it seems that core language UB can't be avoided in the current strategy (for which I submitted [CWG2514](https://cplusplus.github.io/CWG/issues/2514.html)). We need to do different operations in constant evaluation. (No change for this is requested for this PR.)

Now this PR assumes that `pair<K, V>` and `pair<const K, V>` have the same layout, which is not guaranteed - especially when there's a user-defined `pair` specialization. Is it possible to use less scary `const_cast`?

Reply:
We already assume that. No, it's not. We're type-punning and not just playing with constness.

Reply:
Sorry, I might have misunderstood you. Are you suggesting that I should assign the values individually instead of doing the `reinterpret_cast`?

Reply:
> Sorry, I might have misunderstood you. Are you suggesting that I should assign the values individually instead of doing the `reinterpret_cast`?

Yes. I'm afraid this might be necessary somewhere. It seems that the assumption about layout sameness can only be made when node handles are involved (per [[container.node.overview]/4](https://eel.is/c++draft/container.node.overview#4)).

Given that MSVC STL is doing similar `reinterpret_cast` in assignment operators of unordered associative containers, do we want an LWG issue to allow the assumption to be made elsewhere?